server/chat/backend/agent/tools/mcp_tools.py

```
@@ -735,8 +735,13 @@ def get_credentials_for_server(self, server_type: str, user_id: str = None) -> D
             # Get user credentials from Aurora's database
             credentials = get_user_cloud_credentials(user_id)
             if not credentials:
-                logging.warning(f" No credentials found for user {user_id}")
-                return {}
+                # App-mode github users have no entries in user_tokens.
+                # Don't bail here — the github branch below will mint an
+                # installation token from the App. Bail only for non-github.
+                if server_type != "github":
+                    logging.warning(f" No credentials found for user {user_id}")
```

Ruff (0.15.12)

[warning] Logging statement uses f-string

(G004)